core/src/main/scala/org/http4s/Message.scala

```diff
@@ -586,6 +601,17 @@ final case class Response[F[_]](
 }
 
 object Response {
+  def apply[F[_]](
+      status: Status = Status.Ok,
+      httpVersion: HttpVersion = HttpVersion.`HTTP/1.1`,
+      headers: Headers = Headers.empty,
+      body: EntityBody[F] = EmptyBody,
+      attributes: Vault = Vault.empty): Response[F] =
+    new Response(status, httpVersion, headers, body, attributes)
+
+  def unapply[F[_]](r: Response[F]): Option[(Status, HttpVersion, Headers, EntityBody[F], Vault)] =
```

Comment:
Should this also take a `Message[F]` like `Request`?

Reply:
No, taking a `Response` makes sense to me.  The one on `Request` looks weird.

Reply:
I will change both.